autoload/lsp.vim

```diff
@@ -700,10 +700,16 @@ function! s:on_notification(server_name, id, data, event) abort
 
     if lsp#client#is_server_instantiated_notification(a:data)
         if has_key(l:response, 'method')
-            if g:lsp_diagnostics_enabled && l:response['method'] ==# 'textDocument/publishDiagnostics'
+            let l:method = l:response['method']
+            if g:lsp_diagnostics_enabled && l:method ==# 'textDocument/publishDiagnostics'
                 call lsp#ui#vim#diagnostics#handle_text_document_publish_diagnostics(a:server_name, a:data)
-            elseif l:response['method'] ==# 'textDocument/semanticHighlighting'
+            elseif l:method ==# 'textDocument/semanticHighlighting'
                 call lsp#ui#vim#semantic#handle_semantic(a:server_name, a:data)
+            elseif l:method ==# 'window/logMessage'
+                call lsp#ui#vim#window#log_message(str2nr(l:response['params']['type']), l:response['params']['message'])
```

Comment:
We should remove or disable this line?
I think, `window/logMessage` results is noisy to show.

Reply:
I try to use this with gopls.
A lot of messages shown.

Reply:
Can we remove `window/logMessage`. It is too much noise even for `lsp#log()`

Reply:
It can be super noisy, but I still think `window/logMessage` information needs to be available for people to opt-in to, since some language servers use it for some fairly important information. For example, the Scala language server uses these messages to tell the user that the sbt import has started; without them, it just looks like Vim is sitting there doing nothing.

As an alternative approach that should be less noisy, I updated the implementation to, instead of echoing incoming log messages, set them to a couple global variables: `g:lsp_last_log_message_<type|text>`. The idea is that most users will not see them, but you can opt in to them by including one or both of those variables in your status line, i.e.

```vimscript
set statusline+=%{'['.g:lsp_last_log_message_type.']\ '.g:lsp_last_log_message_text}
```

This way the information does not get lost for those who need it, but it doesn't clog up the logs or the Vim command line. Other editors like VSCode put this information in the status bar, so this is consistent with the experience in other editors as well.